### ISSUE TITLE
Having configured a default group name causes issues when listing virtual machines

### ISSUE THREAD

**Issue by mw007** [Azure/azure-cli#3949]
### Description
If you've configured a default resource group using the `az configure -d group` command, and that resource group no longer exists (e.g. it was deleted), then subsequent calls to `az vm list` will fail with the following error:

~~~
ERROR: Resource group '<name>' could not be found.
~~~

### How to reproduce the issue

1. Configure your default group with a non-existent resource group name: `az configure -d group='groupNotThere'
- Issue an `az vm list` command

### Resolution

This can be fixed by doing `az configure -d group=''` to clear the default group, but perhaps a more descriptive error could be displayed when this condition occurs?

---

### Environment summary

**Install Method:** docker

**CLI Version:** What version of the CLI and modules are installed? (Use `az --version`)  
~~~
# az --version
azure-cli (2.0.2+dev)

acr (2.0.0+dev)
acs (2.0.2+dev)
appservice (0.1.2+dev)
batch (2.0.0+dev)
cloud (2.0.0+dev)
component (2.0.0+dev)
configure (2.0.2+dev)
container (0.1.2+dev)
core (2.0.2+dev)
dla (0.0.1+dev)
dls (0.0.1+dev)
documentdb (0.1.2+dev)
feedback (2.0.0+dev)
find (0.0.1b1+dev)
iot (0.1.2+dev)
keyvault (2.0.0+dev)
lab (0.0.1+dev)
monitor (0.0.1+dev)
network (2.0.2+dev)
nspkg (2.0.0+dev)
profile (2.0.2+dev)
redis (0.1.1b3+dev)
resource (2.0.2+dev)
role (2.0.1+dev)
sql (2.0.0+dev)
storage (2.0.2+dev)
taskhelp (0.1.1b4+dev)
vm (2.0.2+dev)

Python (Linux) 3.5.2 (default, Dec 27 2016, 21:33:11) 
[GCC 5.3.0]
~~~

**OS Version:** What OS and version are you using?  
Red Hat Enterprise Linux Server release 7.2 (Maipo)

**Shell Type:** What shell are you using? (e.g. bash, cmd.exe, Bash on Windows)  
bash

**Comment by yugangw-msft** [comment]
Non mainline scenario. Easy to clean up.

**Comment by haokanga** [comment]
I do not agree if it is fair to assume that this is "non-mainline" scenario -- for beginners who have a minimal CLI knowledge, we are recommended them to use Azure Cloud Shell in the course work. 

In pedagogical settings, when a student sets a default group in one assignment, it can easily cause trouble in other assignments, which hurts student user experience with Azure badly. 

**When Resource group is not specified and Azure CLI fall back to the default resource group, Azure CLI should be transparent and tell the users so**. e.g., 

"You did not specify a resource group and the default resource group X is used. If you want to execute the commands in the scope of all the resource groups in the subscription, use `az configure -d group=''` to unset the default resource group."

**Comment by bureado** [comment]
After talking to @haokanga and the CMU team, here's my understanding of this issue.

In the [az vm](https://docs.microsoft.com/en-us/cli/azure/vm?view=azure-cli-latest#az-vm-list) docs we say today that `az vm list` will **List all VMs**.

However, @haokanga is not seeing that behavior. What he and the students are seeing is a list of VMs _for the default RG_. If the default RG isn't used much, then this command will likely return empty which gives a false impression of no VMs in the subscription.

Here are some of the options that @haokanga and team believe could add more transparency for this behavior:

1. Emit a warning message (such as "Only showing resources in RG $foo") when `az vm` goes through the "default RG" logic path
2. Make `-g` a mandatory argument for `az vm list`, either by encouraging the user to be explicit about the resource group name, or also by permitting a `--default` or `-d` parameter in lieu of naming the default RG
3. Introduce a `--all` option that is a real "list all my VMs" option

@haokanga also had some ideas about bounding `az configure -d group` to some lifecycle event (Cloud Shell sessions, etc.) so I'll let him elaborate on that.

Overall I think the issue here is that `az vm list` is promising a comprehensive listing of VMs but in fact that might not always be the actual behavior.
